### PR TITLE
Added RetrofitSingleton

### PR DESCRIPTION
**Pull request by epicadk**
### Description

Interceptor for Header, RxJava3 Call Adapter Factory dependency added

### Changes Done

No screenshots

### Acknowledgements

Dependencies 
Please ensure the following and put a check accordingly

- [✔] My code doesn't break/affect any unrelated part of the codebase

- [✔] Squashed multiple commits (_if irrelevant_) into one commit

### CODE REVIEW

File: app/src/main/java/com/tip/lunchbox/data/util/RetrofitSingleton.java [hunk 1]
```diff
@@ -0,0 +1,36 @@
+package com.tip.lunchbox.data.util;
+
+import okhttp3.OkHttpClient;
+import retrofit2.Retrofit;
+import retrofit2.adapter.rxjava3.RxJava3CallAdapterFactory;
+import retrofit2.converter.gson.GsonConverterFactory;
+
+public class RetrofitSingleton {
```
Comment (devansh-299) [inline app/src/main/java/com/tip/lunchbox/data/util/RetrofitSingleton.java]: Nice work! but here are few things that need attention:
* The class name should follow the pattern `%Util`. So in this case, ideally the class should be named `RetrofitUtil`
* Util classes should be declared _final_
* All variables and methods should ideally be _static_.

**Review by devansh-299 — CHANGES_REQUESTED** [review]

File: app/build.gradle [hunk 2]
```diff
@@ -44,6 +44,9 @@ dependencies {
     implementation "io.reactivex.rxjava3:rxjava:$rootProject.rx_java_version"
     implementation 'androidx.legacy:legacy-support-v4:1.0.0'
 
+    //Retrofit to RxJava CallAdapter Factory
+    implementation 'com.squareup.retrofit2:adapter-rxjava3:2.9.0'
```
Comment (devansh-299) [inline app/build.gradle]: also here you can directly use `$rootProject.retrofit_version` to maintain uniformity in the codebase

**Review by devansh-299 — CHANGES_REQUESTED** [review]

File: app/build.gradle (hunk 2 above)
Comment (epicadk) [reply app/build.gradle]: Done

File: app/src/main/java/com/tip/lunchbox/data/util/RetrofitSingleton.java (hunk 1 above)
Comment (epicadk) [reply app/src/main/java/com/tip/lunchbox/data/util/RetrofitSingleton.java]: Check now

**Comment by devansh-299** [comment]
@epicadk looks good to me, can you try to squash the commits into one single commit? After that, I can merge the pull request.

**Comment by epicadk** [comment]
> @epicadk looks good to me, can you try to squash the commits into one single commit? After that, I can merge the pull request.

Done

**Pull request merged**